### PR TITLE
feat(wallet) use sql activity filter for incremental updates

### PR DESCRIPTION
**Pull request by stefandunca**
### Updates status-desktop [#12120](https://github.com/status-im/status-desktop/issues/12120)

Switch from the prototype of duplicating the SQL filter as a runtime and keeping them in sync on each event that might invalidate the current filtered entries to a simpler approach of requesting the filter again and doing the diff to detect the new changes.

Also, add a new reset API to model the new entries design requirements.

The new approach shows fewer corner cases to handle and follows one source of truth, making debugging and future maintenance easier.

Other changes

- Fix pending mocking to work with multiple calls
- Refactor tests to account for the new changes

### CODE REVIEW

File: services/wallet/activity/session.go [hunk 1]
```diff
@@ -18,12 +17,28 @@ import (
...
 
+const nilStr = "nil"
+
 type EntryIdentity struct {
 	payloadType PayloadType
 	transaction *transfer.TransactionIdentity
 	id          transfer.MultiTransactionIDType
 }
 
+// func (e EntryIdentity) same(a EntryIdentity) bool {
```
Comment (dlipicar) [inline services/wallet/activity/session.go:28]: can this be removed?

Comment (stefandunca) [reply services/wallet/activity/session.go:28]: I revived and used it in one of my following PRs :)

**Review by dlipicar — APPROVED** [review]

File: services/wallet/activity/session.go
Comment (Cuteivist) [inline services/wallet/activity/session.go:1]: There's a lot of commented functions here. Is there purpose to leave them right now ?

**Review by Cuteivist — APPROVED** [review]
Looks good

File: services/wallet/activity/session.go
Comment (Cuteivist) [reply services/wallet/activity/session.go:1]: Nvm I see they are removed in next PR
